### PR TITLE
Add hightouch tasks

### PR DESCRIPTION
**Pull request by AlessandroLollo**
## Summary
This PR introduces the ability trigger sync runs on Hightouch.

## Importance
Now users can trigger Hightouch Syncs with a native Prefect task.

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: tests/tasks/hightouch/test_hightouch_tasks.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        )
+
+        response = ht_task.run(api_key="key", sync_id=123, wait_for_completion=True)
+
+        assert (
+            responses.assert_call_count(
+                f"https://api.hightouch.io/api/v2/rest/run/{sync_id}", 1
+            )
+            is True
+        )
```
Comment (desertaxle) [inline tests/tasks/hightouch/test_hightouch_tasks.py:127]: This should raise an `AssertionError` if the call count doesn't match.
```suggestion
            responses.assert_call_count(f"https://api.hightouch.io/api/v2/rest/run/{sync_id}", 1)
```

File: tests/tasks/hightouch/test_hightouch_tasks.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        assert (
+            responses.assert_call_count(
+                f"https://api.hightouch.io/api/v2/rest/sync/{sync_id}", 1
+            )
+            is True
+        )
```
Comment (desertaxle) [inline tests/tasks/hightouch/test_hightouch_tasks.py:133]:
```suggestion
        responses.assert_call_count(f"https://api.hightouch.io/api/v2/rest/sync/{sync_id}", 1)
```

**Review by desertaxle — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Overall, it looks great. My comments mostly center around docstrings and tests.

File: tests/tasks/hightouch/test_hightouch_tasks.py (hunk 2 above)
Comment (AlessandroLollo) [reply tests/tasks/hightouch/test_hightouch_tasks.py:133]: I tried your suggeston, and Black rewrites it as the original code.

File: src/prefect/tasks/hightouch/hightouch_tasks.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        self,
+        api_key: str = None,
+        api_key_env_var: str = None,
+        sync_id: int = None,
+        wait_for_completion: bool = False,
+        wait_time_between_api_calls: int = 10,
+        max_wait_time: int = None,
+    ):
+        """
+        This task can be used to trigger a new sync run on [Hightouch]().
```
Comment (tpdorsey) [inline src/prefect/tasks/hightouch/hightouch_tasks.py:73]: URL for Hightouch missing in the link here

File: src/prefect/tasks/hightouch/hightouch_tasks.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        Under the hood it uses [Hightouch official APIs]().
```
Comment (tpdorsey) [inline src/prefect/tasks/hightouch/hightouch_tasks.py:74]: Missing URL in link here, too

File: src/prefect/tasks/hightouch/hightouch_tasks.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            - api_key_env_var (str, optional): The name of the environment variable that contains
+                the API key to use to authenticate on Hightouch.
+                This value is considered only if `api_key` is not provided.
+            - sync_id (int, optional): The ID of the Hightouch sync to run.
+            - wait_for_completion: (int, optional): Whether to wait for the sync run to finish execution
+                or not.
+                Default to `False`.
+            - wait_time_between_api_calls (int, optional): The number of seconds to
+                wait between API calls.
+                Default to 10.
```
Comment (tpdorsey) [inline src/prefect/tasks/hightouch/hightouch_tasks.py:87]:
```suggestion
                Default is 10.
```

File: src/prefect/tasks/hightouch/hightouch_utils.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        sync_id: int,
+        wait_for_completion: bool,
+        wait_time_between_api_calls: int,
+        max_wait_time: int,
+    ) -> Dict:
+        """
+        Start a new sync run.
+        Optionally, wait for run completion.
+        The sync run is triggered via the
+        [Start new sync run API](https://hightouch.io/docs/syncs/api/#start-a-new-sync-run)
```
Comment (tpdorsey) [inline src/prefect/tasks/hightouch/hightouch_utils.py:63]:
```suggestion
        [Start new sync run API](https://hightouch.io/docs/syncs/api/#start-a-new-sync-run).
```

File: src/prefect/tasks/hightouch/hightouch_utils.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+        Args:
+            - sync_id (int): The sync identifier
```
Comment (tpdorsey) [inline src/prefect/tasks/hightouch/hightouch_utils.py:66]:
```suggestion
            - sync_id (int): The sync identifier.
```

**Review by tpdorsey — COMMENTED** [review]
Looks good! Some minor docstring fixes.

File: tests/tasks/hightouch/test_hightouch_tasks.py (hunk 2 above)
Comment (madkinsz) [reply tests/tasks/hightouch/test_hightouch_tasks.py]: Did you misinterpret this suggestion? `black` shouldn't change the meaning of code which this definitely does.

Comment (AlessandroLollo) [reply tests/tasks/hightouch/test_hightouch_tasks.py]: I think I didn't get the suggestion right...
In your opinion, what would be the most appropriate assertion to write here?

**Review by desertaxle — APPROVED** [review]
LGTM!

**Review by tpdorsey — APPROVED** [review]

**Pull request merged**